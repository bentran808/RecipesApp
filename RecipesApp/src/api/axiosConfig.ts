import axios from 'axios';

export const axiosInstance = axios.create({
  baseURL: 'http://10.0.3.2:3000', //Change to http://10.0.2.2 for AVD and http://localhost for iOS
  headers: {
    'Content-Type': 'application/json'
  }
});
