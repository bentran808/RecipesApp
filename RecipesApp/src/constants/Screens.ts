export default {
  Main: { name: 'Main' },
  Home: { name: 'Home', label: 'HOME' },
  Categories: { name: 'Categories', label: 'CATEGORIES' },
  Recipe: { name: 'Recipe' },
  RecipesList: { name: 'RecipesList' },
  Ingredient: { name: 'Ingredient' },
  IngredientsDetails: { name: 'IngredientsDetails' },
  Search: { name: 'Search', label: 'SEARCH' }
};
