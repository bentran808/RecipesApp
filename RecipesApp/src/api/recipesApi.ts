import { axiosInstance } from './axiosConfig';

export const recipesApi = {
  fetchCategoriesRequest() {
    return axiosInstance.get('categories?_embed=recipes');
  },
  fetchIngredientsRequest() {
    return axiosInstance.get('ingredients');
  },
  fetchRecipesRequest() {
    return axiosInstance.get('recipes?_expand=category');
  },
  searchByCategoryNameRequest(keyword: string) {
    return axiosInstance.get(`categories?name_like=${keyword}&_embed=recipes`);
  },
  searchByRecipeNameRequest(keyword: string) {
    return axiosInstance.get(`recipes?title_like=${keyword}`);
  }
};
