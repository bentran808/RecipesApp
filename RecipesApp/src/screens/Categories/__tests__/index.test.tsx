import { render } from '@testing-library/react-native';
import { recipesApi } from 'api';
import Screens from 'constants/Screens';
import { category } from 'mocks';
import React from 'react';
import { Alert, TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';
import CategoriesScreen from '..';

describe('Categories Screen', () => {
  let navigation: any;

  beforeEach(() => {
    navigation = {
      navigate: jest.fn(),
      setOptions: jest.fn(),
      openDrawer: jest.fn()
    };
    const setCategories = jest.fn();
    jest.spyOn(React, 'useEffect').mockImplementation((f) => f());
    React.useState = jest.fn().mockReturnValue([[category], setCategories]);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should render correctly', () => {
    const tree = renderer.create(<CategoriesScreen navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call function handlePressCategory', () => {
    const component = renderer.create(<CategoriesScreen navigation={navigation} />);
    const button = component.root.findAllByType(TouchableOpacity)[0];

    button.props.onPress();

    expect(navigation.navigate).toHaveBeenCalledWith(Screens.RecipesList.name, { category });
  });

  test('should call function recipesApi.fetchCategoriesRequest successful', () => {
    recipesApi.fetchCategoriesRequest = jest.fn().mockReturnValue([category]);

    render(<CategoriesScreen navigation={navigation} />);

    expect(recipesApi.fetchCategoriesRequest).toHaveBeenCalled();
  });

  test('should call function recipesApi.fetchCategoriesRequest failed', () => {
    jest.spyOn(Alert, 'alert');
    recipesApi.fetchCategoriesRequest = jest.fn().mockImplementation(() => {
      throw new Error('Network Error');
    });

    render(<CategoriesScreen navigation={navigation} />);

    expect(Alert.alert).toHaveBeenCalledWith('Fetch data failed');
  });
});
