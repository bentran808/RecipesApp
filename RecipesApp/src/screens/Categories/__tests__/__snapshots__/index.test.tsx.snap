// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Screen should render correctly 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 0,
          "name": "Cookies",
          "photo_url": "https://www.telegraph.co.uk/content/dam/Travel/2019/January/france-food.jpg?imw'id'th=1400",
          "recipes": Array [],
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#ccc",
                "borderRadius": 20,
                "borderWidth": 0.5,
                "height": 215,
                "justifyContent": "center",
                "margin": 10,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://www.telegraph.co.uk/content/dam/Travel/2019/January/france-food.jpg?imw'id'th=1400",
                }
              }
              style={
                Object {
                  "borderBottomLeftRadius": 0,
                  "borderBottomRightRadius": 0,
                  "borderRadius": 20,
                  "height": 155,
                  "shadowColor": "blue",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 0,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 5,
                  "width": "100%",
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#333",
                  "flex": 1,
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "marginTop": 8,
                }
              }
            >
              Cookies
            </Text>
            <Text
              style={
                Object {
                  "marginBottom": 5,
                  "marginTop": 3,
                }
              }
            >
              0
               recipes
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
